Libraries/JavaScriptAppEngine/System/JSTimers/JSTimersExecution.js

```diff
@@ -108,6 +117,34 @@ var JSTimersExecution = {
     }
   },
 
+  callIdleCallbacks: function(frameTime) {
+    // Arbitrary threshold: 10ms remaining in frame.
+    if (17 - (performanceNow() - frameTime) < 10) {
```

Comment:
Note that the new React Fiber reconciler for DOM does __all__ reconciliation except for animations using requestIdleCallback. https://github.com/facebook/react/blob/cf259a4ff82bb6248b41392e821d2822f9ce46c6/src/renderers/shared/fiber/ReactFiberScheduler.js#L199

So it is very important that it gets enough cycles. It is a misnomer to think that it is only for background work.